tests/pipelines/test_step_utils.py

```diff
@@ -75,7 +75,7 @@ def test_get_merged_eval_metrics_works(
         (DataFrame(np.arange(160).reshape(8, 20)), 1000, 20, 8, 20, 8),
         (DataFrame(np.arange(160).reshape(8, 20)), 1000, 21, 8, 20, 8),
         (DataFrame(np.arange(160).reshape(8, 20)), 1000, 20, 9, 20, 8),
-        (DataFrame(np.arange(160).reshape(8, 20)), 10, 5, 9, 5, 2),
+        (DataFrame(np.arange(160).reshape(8, 20)), 10, 5, 9, 5, 8),
```

Comment:
In this case, the expected_rows * expected_columns  exceeds max_cells, so I don't think the adjustment to https://github.com/mlflow/mlflow/pull/6297/files#diff-e02b9988953b168d7b15748927274c8a44225b3caeb346ce655b90b79c4aa815R125 is correct

Reply:
Basically I was trying to solve for the case where

we have 2000 cells, 1000 columns and 2row.
Now the Max is 1000 cells, 100 columns.  Based on the old math, it would be 10rows in the sample. The number of available rows are 2. So sampling 10rows would be a lot and incorrect. 

That caused me to change the above math.


Now in the new way, I added:

```
max_rows = min(
        max(max_cells // max_cols, 1),
        (data_frame.size // data_frame.columns.size),
        _MAX_PROFILE_ROW_SIZE,
    )
    ```
    
    Which should hopefully catch the other usecase as well. Let me add a test for what I was saying above.